test/internal/helpers/Player.java

```diff
@@ -440,32 +1031,79 @@ public static Cleanups withItemMonster(final String itemMonster) {
         });
   }
 
-  public static Cleanups canInteract(boolean canInteract) {
+  /**
+   * Sets the player's "can interact" status (i.e. whether they are under ronin-style restrictions)
+   *
+   * @param canInteract Whether the player can interact
+   * @return Restores to previous value
+   */
+  public static Cleanups withInteractivity(boolean canInteract) {
     var old = CharPaneRequest.canInteract();
     CharPaneRequest.setCanInteract(canInteract);
     return new Cleanups(() -> CharPaneRequest.setCanInteract(old));
   }
 
+  /**
+   * Sets the player's standard restriction status
+   *
+   * @param restricted Whether the player is standard restricted
+   * @return Restores to previous value
+   */
+  public static Cleanups withRestricted(boolean restricted) {
+    var old = KoLCharacter.getRestricted();
+    KoLCharacter.setRestricted(restricted);
+    return new Cleanups(() -> KoLCharacter.setRestricted(old));
+  }
+
+  /**
+   * Sets the basement level
+   *
+   * @param level Level to set
+   * @return Restores to the previous value
+   */
   public static Cleanups withBasementLevel(final int level) {
     var old = BasementRequest.getBasementLevel();
     BasementRequest.setBasementLevel(level);
     return new Cleanups(() -> BasementRequest.setBasementLevel(old));
   }
 
+  /**
+   * Sets the basement level to zero This is useful if the tested function is going to set the level
+   * and we just want to make sure it gets reset
+   *
+   * @return Restores the previous value
+   */
   public static Cleanups withBasementLevel() {
     return withBasementLevel(0);
   }
 
+  /**
+   * Sets the continuation state (i.e. whether the CLI is in green "good", red "bad" mode etc)
+   *
+   * @param continuationState State to set
+   * @return Restores previous continuation state
+   */
   public static Cleanups withContinuationState(final KoLConstants.MafiaState continuationState) {
```

Comment:
Personal preference:

import net.sourceforge.kolmafia.KoLConstants.MafiaState;

And just MafiaState.xxx everywhere.